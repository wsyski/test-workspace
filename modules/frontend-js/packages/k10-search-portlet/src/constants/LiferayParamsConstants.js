export const VIEW_MODE_FULL = 'FULL';
export const VIEW_MODE_BRIEF = 'BRIEF';

export const BASE_URL = 'baseUrl';
export const SEARCH_PAGE = 'searchPage';
export const SEARCH_QUERY_PARAMETER = 'searchQueryParameter';
export const COLLECTION = 'collection';
export const PAGE_SIZE = 'pageSize';
export const VIEW_MODE = 'viewMode';

export const PORTLET_INSTANCE_DEFAULT = {
	[BASE_URL]: 'https://bookit.axiell.com/k10/index/axiell',
	[COLLECTION]: [
		'medline',
		'NL',
		'JSTOR',
		'springer',
		'DOAJ',
		'elsevier',
		'CTG',
	],
	[PAGE_SIZE]: 8,
	[SEARCH_PAGE]: 'k10-search',
	[SEARCH_QUERY_PARAMETER]: 'p_r_p_arena_urn:arena_search_query',
	[VIEW_MODE]: VIEW_MODE_FULL,
};
