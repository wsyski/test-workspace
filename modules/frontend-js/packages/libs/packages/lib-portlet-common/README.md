# react-portlet

lib-portlet-common
https://dev.to/siddharthvenkatesh/component-library-setup-with-react-typescript-and-rollup-onj
https://github.com/sidv93/react-ts-lib
https://github.com/boilertown/react-ui-boilerplate
