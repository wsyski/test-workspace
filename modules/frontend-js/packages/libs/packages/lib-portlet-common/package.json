{
  "name": "@arena/lib-portlet-common",
  "version": "1.0.0",
  "description": "Portlet common library",
  "license": "UNLICENSED",
  "private": true,
  "main": "dist/cjs/bundle.js",
  "module": "dist/esm/bundle.js",
  "types": "dist/index.d.ts",
  "files": [
    "dist"
  ],
  "source": "src/index.ts",
  "engines": {
    "node": ">=20",
    "yarn": "1.x"
  },
  "dependencies": {
    "@arena/portal": "*",
    "@liferay/portal-7.4": "*"
  },
  "scripts": {
    "deploy": "true",
    "clean": "rimraf dist && rimraf out-tsc",
    "build": "cpx \"../../../../../../scripts/rollup.config.mjs\" . && rollup --config rollup.config.mjs",
    "build:prod": "cross-env NODE_ENV=production && run-s build",
    "test": "jest --verbose",
    "tsc": "tsc -b tsconfig.json",
    "publish": "true"
  }
}
