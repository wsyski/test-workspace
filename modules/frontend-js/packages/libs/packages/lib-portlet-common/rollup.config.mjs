import babel from '@rollup/plugin-babel';
import commonjs from "@rollup/plugin-commonjs";
import terser from '@rollup/plugin-terser';
import { readFile } from 'fs/promises';

const packageJson = JSON.parse(await readFile(new URL('./package.json', import.meta.url)));
const outputOptions = {
    interop: 'auto',
    sourcemap: true,
    banner: `/*
 * Arena portlet common library
 * {@link https://github.com/wsyski/test-workspace}
 * @copyright Wojciech Syski (@wsyski)
 * @license MIT
 */`,
};
const input = 'out-tsc/index.js';
const external = ['node_modules', ...Object.keys(packageJson.dependencies || {}), ...Object.keys(packageJson.peerDependencies || {})];

const getPlugins = (format = 'esm') => {

    return [
        commonjs(),
        babel({
            babelHelpers: 'runtime',
            exclude: /node_modules/,
            plugins: [['@babel/plugin-transform-runtime'], ["@babel/plugin-proposal-nullish-coalescing-operator"], ["@babel/plugin-proposal-class-properties"]],
            presets: [
                ['@babel/preset-env'],
                ['@babel/preset-react'],
            ],
        }),
        (process.env.NODE_ENV === 'production' && terser())
    ];
};

const getOutput = (format = 'esm') => {
    return {
        dir: 'dist/'+format,
        format,
        ...outputOptions
    };
}

export default [
    // cjs configuration
    {
        input,
        output: getOutput('cjs'),
        plugins: getPlugins('cjs'),
        external,
    },
    // esm configuration
    {
        input,
        output: getOutput('esm'),
        plugins: getPlugins('esm'),
        external,
    }
];
