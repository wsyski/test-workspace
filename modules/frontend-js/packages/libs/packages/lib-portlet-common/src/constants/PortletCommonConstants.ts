import {LocalizationParams} from '../index';

export const HTTP_STATUS_BAD_REQUEST = 400;
export const HEADER_ACCEPT_LANGUAGE = 'Accept-Language';
export const HEADER_CONTENT_TYPE = 'Content-Type';
export const HEADER_AUTHORIZATION = 'Authorization';

export const ALERT_ORIGIN_DEFAULT = 'global';
export const ERRORS_PREFIX = 'errors.';
export const PROJECT_VERSION = '5.8.0';

export const LOCALIZATION_PARAMS_DEFAULT = {
	dateFormat: {
		DAYS: 'yyyy-MM-dd',
		MONTHS: 'yyyy-MM',
		SECONDS: 'yyyy-MM-dd HH:mm',
		YEARS: 'yyyy',
	},
	firstDayOfWeek: 1,
	months: [
		'January',
		'February',
		'March',
		'April',
		'May',
		'June',
		'July',
		'August',
		'September',
		'October',
		'November',
		'December',
	],
	weekdaysShort: ['S', 'M', 'T', 'W', 'T', 'F', 'S'],
} as LocalizationParams;
