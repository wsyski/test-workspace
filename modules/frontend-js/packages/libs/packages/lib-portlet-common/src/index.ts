/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

import {DisplayType} from '@clayui/alert';
import {FirstDayOfWeek} from '@clayui/date-picker';
import {FormikValues} from 'formik';

import AlertMessages from './components/AlertMessages';
import ArenaIcon, {
	ARENA_ICON_SPRITEMAP,
	ArenaIconSpriteContext,
} from './components/ArenaIcon';
import ErrorBoundary from './components/ErrorBoundary';
import LoadingIndicator from './components/LoadingIndicator';
import Permalink from './components/Permalink';
import ShowMoreList from './components/ShowMoreList';
import ShowMoreText from './components/ShowMoreText';
import ErrorFocus from './components/form/ErrorFocus';
import {
	ErrorFeedback,
	HelpMessage,
	RequiredMark,
	SelectOptions,
	getFieldFeedbackId,
} from './components/form/FormInput';
import {
	ALERT_ORIGIN_DEFAULT,
	ERRORS_PREFIX,
	HEADER_AUTHORIZATION,
	HTTP_STATUS_BAD_REQUEST,
	LOCALIZATION_PARAMS_DEFAULT
} from './constants/PortletCommonConstants';
import AlertActionsContext from './contexts/AlertActionsContext';
import CommonServicesConfigContext, {
	COMMON_SERVICES_CONFIG_DEFAULT,
} from './contexts/CommonServicesConfigContext';
import LiferayParamsContext, {
	LIFERAY_PARAMS_DEFAULT,
} from './contexts/LiferayParamsContext';
import useLiferayParams from './hooks/useLiferayParams';
import useLocalStorage from './hooks/useLocalStorage';
import useSessionStorage from './hooks/useSessionStorage';
import EventSourceCloseToken from './models/EventSourceCloseToken';
import {AxCurrencyAmount} from './models/domain/ax/AxCurrencyAmount';
import {AxExternalUrl} from './models/domain/ax/AxExternalUrl';
import {AxImageUrl} from './models/domain/ax/AxImageUrl';
import CommonServicesConfigService from './services/CommonServicesConfigService';
import AlertUtil from './utils/AlertUtil';
import EventSourceUtil from './utils/EventSourceUtil';
import I18nUtil from './utils/I18nUtil';
import LiferayUtil from './utils/LiferayUtil';
import MiscUtil from './utils/MiscUtil';
import RoutingUtil from './utils/RoutingUtil';
import ServiceUtil from './utils/ServiceUtil';
import SuspensibleResource from './utils/SuspensibleResource';

export type FieldValidations<V extends FormikValues> = {
	[K in keyof V]?: ((value: any) => string | undefined)[];
};

export type AlertConfirmType = 'ToastConfirm' | 'ModalConfirm'| 'ModalConfirmCancel';

export interface Alert {
	message: string;
	displayType: DisplayType;
	origin: string;
	confirmType: AlertConfirmType;
	title?: string;
}

export interface LiferayParamsConfiguration {
	portletInstance: Configuration;
	system: Configuration;
}

export interface LiferayParams {
	contextPath: string;
	portletElementId: string;
	portletNamespace: string;
	configuration: LiferayParamsConfiguration;
}

export interface AlertActions {
	addAlert: (alert: Alert) => void;
	deleteAlert: (alert: Alert) => void;
	error: (message: string) => void;
	warning: (message: string) => void;
	info: (message: string) => void;
	success: (message: string) => void;
}

export interface AppContainerProps {
	liferayParams: LiferayParams;
}

export interface Configuration {
	[key: string]: string | string[];
}

export type MappedPropertyType<T, F extends keyof T, U> = {
	[K in keyof T]: K extends F ? U : T[K];
};

export interface ExtendedLiferayParams extends LiferayParams {
	getSystemValueAsArray: (key: string) => string[];
	getSystemValueAsBoolean: (key: string) => boolean;
	getSystemValueAsNumber: (key: string) => number;
	getSystemValueAsString: (key: string) => string;
	getInstanceValueAsArray: (key: string) => string[];
	getInstanceValueAsBoolean: (key: string) => boolean;
	getInstanceValueAsNumber: (key: string) => number;
	getInstanceValueAsString: (key: string) => string;
}

export interface LocalizationParams {
	currencyAmount: string;
	dateFormat: string;
	firstDayOfWeek: FirstDayOfWeek;
	months: string[];
	weekdaysShort: string[];
}

export interface CommonServicesConfig {
	calendarApiEndpoint: string;
	calendarCustomerId: string;
	calendarDefaultAllowedAttendees: number;
	calendarLocationVocabularyId: number;
	calendarTargetAudienceVocabularyId: number;
	federatedSearchCustomerId: string;
	federatedSearchApiEndpoint: string;
	federatedSearchSourceConfig: any;
	googleAnalyticsMeasurementId: string;
	openingHoursApiEndpoint: string;
	openingHoursCustomerId: string;
	transactionApiEndpoint: string;
	transactionTenantId: string;
}

export {
	ARENA_ICON_SPRITEMAP,
	ALERT_ORIGIN_DEFAULT,
	COMMON_SERVICES_CONFIG_DEFAULT,
	ERRORS_PREFIX,
	LIFERAY_PARAMS_DEFAULT,
	HTTP_STATUS_BAD_REQUEST,
	HEADER_AUTHORIZATION,
	LOCALIZATION_PARAMS_DEFAULT,
	AlertActionsContext,
	AlertUtil,
	ArenaIcon,
	ArenaIconSpriteContext,
	AlertMessages,
	CommonServicesConfigService,
	CommonServicesConfigContext,
	ErrorFocus,
	ErrorBoundary,
	EventSourceUtil,
	EventSourceCloseToken,
	ErrorFeedback,
	HelpMessage,
	RequiredMark,
	SelectOptions,
	I18nUtil,
	LiferayUtil,
	LiferayParamsContext,
	LoadingIndicator,
	MiscUtil,
	Permalink,
	RoutingUtil,
	ServiceUtil,
	ShowMoreList,
	ShowMoreText,
	SuspensibleResource,
	getFieldFeedbackId,
	useLiferayParams,
	useLocalStorage,
	useSessionStorage
};

export type {AxCurrencyAmount, AxExternalUrl, AxImageUrl};
