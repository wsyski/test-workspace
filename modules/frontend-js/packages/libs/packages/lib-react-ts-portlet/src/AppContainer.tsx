/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

import {
	ALERT_ORIGIN_DEFAULT,
	Alert,
	AlertActions,
	AlertActionsContext,
	AlertMessages,
	AlertUtil,
	AppContainerProps,
	ErrorBoundary,
	LiferayParamsContext,
	LiferayUtil,
} from '@arena/lib-portlet-common';
import {ClayIconSpriteContext} from '@clayui/icon';
import React, {Suspense, useImperativeHandle, useRef} from 'react';

import App from './components/App';

interface AlertPanelProps {}

const AlertPanel = React.forwardRef(
	(props: AlertPanelProps, ref: React.Ref<AlertActions>) => {
		const [alerts, setAlerts] = React.useState<Alert[]>([]);

		const alertActions = AlertUtil.getAlertActions(setAlerts);

		useImperativeHandle(ref, () => alertActions);

		const onConfirm = React.useCallback(
			(alert: Alert) => {
				switch (alert.origin) {
					case ALERT_ORIGIN_DEFAULT:
						alertActions.deleteAlert(alert);
						break;
					default:
						throw new Error(
							'Unexpected alert origin: ' + alert.origin
						);
				}
			},
			[alertActions]
		);

		return <AlertMessages alerts={[...alerts]} onConfirm={onConfirm} />;
	}
);

const AppContainer: React.FC<AppContainerProps> = ({
													   liferayParams,
												   }: AppContainerProps) => {
	const alertActionsRef = useRef<AlertActions>(null);

	return (
		<div data-testid="appContainer">
			<ClayIconSpriteContext.Provider
				value={LiferayUtil.getClaySpritemap()}
			>
				<LiferayParamsContext.Provider value={liferayParams}>
					<Suspense fallback={<></>}>
						<AlertPanel ref={alertActionsRef} />

						<ErrorBoundary>
							<AlertActionsContext.Provider
								value={alertActionsRef}
							>
								<App />
							</AlertActionsContext.Provider>
						</ErrorBoundary>
					</Suspense>
				</LiferayParamsContext.Provider>
			</ClayIconSpriteContext.Provider>
		</div>
	);
};

export default AppContainer;
