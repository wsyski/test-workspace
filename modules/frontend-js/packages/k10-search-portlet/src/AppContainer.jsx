import {LiferayParamsContext, LiferayUtil} from '@arena/lib-portlet-common';
import {ClayIconSpriteContext} from '@clayui/icon';
import React from 'react';
import {HashRouter as Router, Route, Switch} from 'react-router-dom';

import App from './App';
import RecordWrapper from './components/RecordWrapper';
import SearchResultWrapper from './components/SearchResultWrapper';

const AppContainer = ({liferayParams}) => {
	return (
		<div data-test-id="appContainer">
			<ClayIconSpriteContext.Provider
				value={LiferayUtil.getClaySpritemap()}
			>
				<LiferayParamsContext.Provider value={liferayParams}>
					<App>
						<Router>
							<Switch>
								<Route
									path="/record/:id"
									render={(props) => {
										return (
											<RecordWrapper
												id={props.match.params.id}
												{...props}
											/>
										);
									}}
								/>

								<Route
									component={SearchResultWrapper}
									exact
									path=""
								/>
							</Switch>
						</Router>
					</App>
				</LiferayParamsContext.Provider>
			</ClayIconSpriteContext.Provider>
		</div>
	);
};

export default AppContainer;
