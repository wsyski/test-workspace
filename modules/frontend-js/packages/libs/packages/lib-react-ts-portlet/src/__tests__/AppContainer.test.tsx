import {LIFERAY_PARAMS_DEFAULT, LiferayUtil} from '@arena/lib-portlet-common';
import {screen} from '@testing-library/dom';
import {render} from '@testing-library/react';
import React from 'react';

import AppContainer from '../AppContainer';
import {PORTLET_INSTANCE_DEFAULT} from '../constants/LiferayParamsConstants';
(function (exports) {
	exports.Language = {
		get(id, args) {
			return args ? id + ' ' + args.join(',') : id;
		},
	};
})(((window as any).Liferay = (window as any).Liferay || {}));


describe('AppContainer Component', () => {

	beforeEach(() => {
		jest.spyOn(LiferayUtil, 'getScopeGroupId').mockImplementation(
			() => 39847
		);
		jest.spyOn(LiferayUtil, 'getPortalURL').mockImplementation(
			() => 'http://localhost:6080'
		);
		jest.spyOn(LiferayUtil, 'getPathThemeImages').mockImplementation(
			() => '/theme/images'
		);
		jest.spyOn(LiferayUtil, 'getLocale').mockImplementation(() => 'en_US');
		jest.spyOn(LiferayUtil, "getLocale").mockImplementation(() => "en_US");

	});

	afterEach(() => {
		jest.clearAllMocks();
	});

	test('AppContainer should mount without errors', () => {
		const liferayParamsWithDefaults = LiferayUtil.setLiferayParamsDefaults(
			LIFERAY_PARAMS_DEFAULT,
			PORTLET_INSTANCE_DEFAULT
		);
		const {container} = render(
			<AppContainer liferayParams={liferayParamsWithDefaults} />
		);

		// screen.debug();

		const elements = screen.getAllByTestId('appContainer');
		expect(elements.length).toBe(1);
		expect(container).toMatchSnapshot();
	});
});
