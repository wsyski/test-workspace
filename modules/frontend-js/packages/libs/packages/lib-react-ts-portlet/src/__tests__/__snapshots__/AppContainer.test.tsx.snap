// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContainer Component AppContainer should mount without errors 1`] = `
<div>
  <div
    data-testid="appContainer"
  >
    <div>
      <div>
        <span
          class="tag"
        >
          portletNamespace.label
          :
        </span>
        <span
          class="value"
        >
          _portlet_namespace_
        </span>
      </div>
      <div>
        <span
          class="tag"
        >
          contextPath.label
          :
        </span>
        <span
          class="value"
        />
      </div>
      <div>
        <span
          class="tag"
        >
          portletElementId.label
          :
        </span>
        <span
          class="value"
        >
          js-portlet-_portlet_namespace_
        </span>
      </div>
      <div>
        <span
          class="tag"
        >
          configuration.label
          :
        </span>
        <span
          class="pre value"
        >
          {
  "portletInstance": {
    "drink": "orange"
  },
  "system": {}
}
        </span>
      </div>
      <div>
        <button
          class="extended-button"
        >
          extendedButton.label
        </button>
      </div>
    </div>
  </div>
</div>
`;
